sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/PubsubToIcebergIT.java

```diff
@@ -0,0 +1,249 @@
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests stream writing to Iceberg with Beam SQL. */
+@RunWith(JUnit4.class)
+public class PubsubToIcebergIT implements Serializable {
+  private static final Schema SOURCE_SCHEMA =
+      Schema.builder().addNullableField("id", INT64).addNullableField("name", STRING).build();
+
+  @Rule public transient TestPipeline pipeline = TestPipeline.create();
+  @Rule public transient TestPubsub pubsub = TestPubsub.create();
+
+  private static final BigqueryClient BQ_CLIENT = new BigqueryClient("PubsubToIcebergIT");
+  static final String DATASET = "sql_pubsub_to_iceberg_it_" + System.nanoTime();
+  static String warehouse;
+  protected static final GcpOptions OPTIONS =
+      TestPipeline.testingPipelineOptions().as(GcpOptions.class);
+  @Rule public TestName testName = new TestName();
+
+  @BeforeClass
+  public static void createDataset() throws IOException, InterruptedException {
+    warehouse =
+        String.format(
+            "%s%s/%s",
+            TestPipeline.testingPipelineOptions().getTempLocation(),
+            IcebergReadWriteIT.class.getSimpleName(),
+            UUID.randomUUID());
+    BQ_CLIENT.createNewDataset(OPTIONS.getProject(), DATASET);
+  }
+
+  private String tableIdentifier;
+  private Map<String, String> icebergConfig;
+
+  @Before
+  public void setup() {
+    tableIdentifier = DATASET + "." + testName.getMethodName();
+    icebergConfig =
+        ImmutableMap.of(
+            "catalog-impl", "org.apache.iceberg.gcp.bigquery.BigQueryMetastoreCatalog",
+            "io-impl", "org.apache.iceberg.gcp.gcs.GCSFileIO",
+            "warehouse", warehouse,
+            "gcp_project", OPTIONS.getProject(),
+            "gcp_region", "us-central1");
+  }
+
+  @AfterClass
+  public static void deleteDataset() {
+    BQ_CLIENT.deleteDataset(OPTIONS.getProject(), DATASET);
+  }
+
+  @Test
+  public void testSimpleInsert() throws Exception {
+    String pubsubTableString =
+        "CREATE EXTERNAL TABLE pubsub_topic (\n"
+            + "event_timestamp TIMESTAMP, \n"
+            + "attributes MAP<VARCHAR, VARCHAR>, \n"
+            + "payload ROW< \n"
+            + "             id BIGINT, \n"
+            + "             name VARCHAR \n"
+            + "           > \n"
+            + ") \n"
+            + "TYPE 'pubsub' \n"
+            + "LOCATION '"
+            + pubsub.topicPath()
+            + "' \n"
+            + "TBLPROPERTIES '{ \"timestampAttributeKey\" : \"ts\" }'";
+    String icebergTableString =
+        "CREATE EXTERNAL TABLE iceberg_table( \n"
+            + "   id BIGINT, \n"
+            + "   name VARCHAR \n "
+            + ") \n"
+            + "TYPE 'iceberg' \n"
+            + "LOCATION '"
+            + tableIdentifier
+            + "' \n"
+            + "TBLPROPERTIES '{ \"triggering_frequency_seconds\" : 10 }'";
```

Comment:
Can I define partition column with transform function for my Iceberg Table ?

Reply:
We don't support creating tables with partition spec right now (tracker #34117)